Baker - Enhancements - User controll over Baker #2592

bob0bob opened the pull request:
I did some searches to see if active work is being done on the Baker and I've suggest these features about 9 months ago as what I thought was a good enhancement.

If there is a procedure on new things and how to go about it. Please let me know. I didn't notice one but I manage there is one and hope this doesn't make people mad. Just trying to make the baker a nicer part of the colony.

This is an enhancement for the Baker. I've added a second gui that the user can select which recipes the baker will make. As you can see from the screen shots it shows everything recipe the baker can make. Click on the items you wish them to make. So you can control what they produce. If you only want cookies, just have the cookies the only item clicked. The baker will only make cookies. I've added Pumpkin Pie since it is a vanilla minecraft item.

So there are 4 recipes for the baker now.

If your town gets over loaded with food, you can also uncheck everything and the baker goes IDLE until there is an item that the baker is allowed to bake. You don't have to reassign them to another job anymore if you don't wish too.

Also, in this version, you have the ability to have one baker make cakes, and another make bread. Each building saves what is made in that building so you can have different buildings making different things.

There is no default recipe the baker will request if items are missing like before. They will find the first recipe that is allowed and request those items and until they receive those items, they will wait for the request to be fulfilled.

If you have more than one item checked for allowed recipes, the baker will just go down the list. The baker doesn't care if recipe 1 is fulfill-able and recipe 2 is not. If recipe 2 is the next item to be produce it will request those items and wait for it.

Unless, your building is higher than level 1. So the baker can make more than one item at a time. Say it was level 3. Which means they can be making 3 things at a time.

I've submitted all the code changes. I hope that you guys will accept the new changes to the baker. I think it still follows the scheme of MineColonies.

bob0bob committed:
Fields out of sync to gui settings

There were some bugs introduced that allow the gui assignment to get out of sync with the data from the NBT.  But there is no way to solve this issue.

I add a check to see if the owner of the field is the current citizen and if not then remove it from the list.

bob0bob committed:
Lumberjack saplings - cosmetic possible

If you add a mod or update a mod after the lumberjack hut is placed there is no way to get a full list of saplings.  It was only done when you place the hut.   So this means you had to break your hut and redo it.
This allows the player to get updated list of saplings.

bob0bob committed:
Revert "Fields out of sync to gui settings"

This reverts commit f84ecf6b68ad6548b143ef0912bd59847e963585.

bob0bob committed:
Revert "Lumberjack saplings - cosmetic possible"

This reverts commit 06a1eda18dfcd6cd6ba60e06184e292ec10fa543.

bob0bob committed:
Merge branch 'version/1.12' of https://github.com/ldtteam/minecolonies into ldtteam-version/1.12

bob0bob committed:
Merge branch 'ldtteam-version/1.12' into version/1.12

bob0bob committed:
Herders don't get any experience #2386

Herders wouldn't receive experience points.  The SkillModifier was not being set and the default is 0.

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592

I did some searches to see if active work is being done on the Baker and I've suggest these features about 9 months ago as what I thought was a good enhancement.

If there is a procedure on new things and how to go about it. Please let me know. I didn't notice one but I manage there is one and hope this doesn't make people mad. Just trying to make the baker a nicer part of the colony.

This is an enhancement for the Baker. I've added a second gui that the user can select which recipes the baker will make. As you can see from the screen shots it shows everything recipe the baker can make. Click on the items you wish them to make. So you can control what they produce. If you only want cookies, just have the cookies the only item clicked. The baker will only make cookies. I've added Pumpkin Pie since it is a vanilla minecraft item.

So there are 4 recipes for the baker now.

If your town gets over loaded with food, you can also uncheck everything and the baker goes IDLE until there is an item that the baker is allowed to bake. You don't have to reassign them to another job anymore if you don't wish too.

Also, in this version, you have the ability to have one baker make cakes, and another make bread. Each building saves what is made in that building so you can have different buildings making different things.

There is no default recipe the baker will request if items are missing like before. They will find the first recipe that is allowed and request those items and until they receive those items, they will wait for the request to be fulfilled.

If you have more than one item checked for allowed recipes, the baker will just go down the list. The baker doesn't care if recipe 1 is fulfill-able and recipe 2 is not. If recipe 2 is the next item to be produce it will request those items and wait for it.

Unless, your building is higher than level 1. So the baker can make more than one item at a time. Say it was level 3. Which means they can be making 3 things at a time.

I've submitted all the code changes. I hope that you guys will accept the new changes to the baker. I think it still follows the scheme of MineColonies.

bob0bob committed:
Merge pull request #4 from ldtteam/version/1.12

gettings latest.

bob0bob committed:
changes to meet suggestions on coding, #2592

changes to meet suggestions on coding, #2592

bob0bob commented:
I've made the changes to the code, but not sure how to do another pull request from my branch.  it shows this is open.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33900 |  4  |  7  |  -3  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10711&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33887 |  5  | 18  | -13  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10724&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592

Changes to formatting

bob0bob commented:
Ok.  I've updated the changes.  I think I got all of them changed, and what I've seen.

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java:
```diff
@@ -90,6 +107,10 @@
     private int ticksPassed = 0;
 
+    
+    
```
javadoc please of this array.

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java:
```diff
@@ -418,11 +465,52 @@ public void putInFurnace(final BlockPos currentFurnace, final BakingProduct baki
     }
 
+    
+    public boolean isRecipeAllowed(final int pos)
```
also javadoc to these two please. All public methods should have javadoc

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java:
```diff
@@ -434,6 +522,26 @@ public View(final ColonyView c, final BlockPos l)
+            MineColonies.getNetwork().sendToServer(new AssignBakerRecipeMessage(this, pos, value, block));
+        	if (pos < recipesAllowed.length)
+			{
+        		recipesAllowed[pos] = value;
```
the formatting seems off here

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/BakerRecipes.java:
```diff
@@ -56,7 +59,15 @@
+        final List<ItemStack> inputPumpkinPie = new ArrayList<>();
+        inputPumpkinPie.add(new ItemStack(Blocks.PUMPKIN.getItemDropped(null, null, 0), 1));
+//        inputPumpkinPie.add(new ItemStack(Items.PUMPKIN_SEEDS, 4));
+        inputPumpkinPie.add(new ItemStack(Items.SUGAR, 1));
```
We can remove this recipe as well then =)

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java:
```diff
@@ -98,6 +98,9 @@
     private int progress = 0;
 
+    
+    private int currentRecipe = -1;
```
also, jdoc

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33884 |  0  |  3  |  -3  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:

bob0bob commented:
Latest check shows failed.   It has one file showing should be final on a variable.  I've checked and it is checked in.   I'm not sure if the merge included this, or the check was done before it got it all checked it.

I verified the code on my branch and it shows it in there.  Is there a way to check what it is showing on the pull request?

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592 #2593

adding missing javadoc function definition

bob0bob commented:
updated missing javadoc definitions for new functions.

bob0bob committed:
Merge branch 'version/1.12' into version/1.12

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33882 |  1  |  3  |  -2  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10750&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33882 |  1  |  3  |  -2  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10750&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33881 |  0  |  1  |  -1  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592

formatting changes.

bob0bob committed:
Merge branch 'version/1.12' of https://github.com/bob0bob/minecolonies into version/1.12

bob0bob commented:
Ok.  This is my first time doing this.  When I do an update on my branch, who do you pulling in new changes when you have a pending one.  my branch doesn't show you can.  It only has a "click to view pending"  PR.

bob0bob committed:
Enhancements - User controll over Baker #2592 #2593

adding a final statement

Raycoms commented:
If you push to the same branch you created the PR from, it pulls in the changes automatically

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33881 |  1  |  1  |  0   |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10795&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592

The request for ingredients didn't show up on the request system,  when it determined it didn't have items to make recipe.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33887 |  6  |  0  |  +6  |
| Errors |  591  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10808&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Baker - Enhancements - User controll over Baker #2592

more code changes to confirm to the formatting style of MC.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33883 |  1  |  5  |  -4  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10821&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Town Hall - Should Save build order. #2595

Building priorities were not being saved to the NBT file.  Onload everything was at default again at 0.   Now the priority is saved so on reload they are in order as the player picked them.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33882 |  0  |  1  |  -1  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
Lumberjack - Not saving cutting down tree correctly. #2596

The readFromNBT for lumberjack building was not reading in the save data on what should be cut or not cut.

Also,  The world reading in tree saplings was not checking to see if the sapling doesn't exist anymore.  If it is turned into air, then ignore it.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
removing unneeded imports

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.block.BlockRedstoneOre;
```
### Errors:
- Unused import import net.minecraft.block.BlockRedstoneOre;

Raycoms replied:
Are these here really unused? I don't trust it =D

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.block.BlockRedstoneOre;
 import net.minecraft.block.state.IBlockState;
```
### Errors:
- Unused import import net.minecraft.block.state.IBlockState;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.block.BlockRedstoneOre;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
```
### Errors:
- Unused import import net.minecraft.creativetab.CreativeTabs;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.block.BlockRedstoneOre;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.init.Items;
```
### Errors:
- Unused import import net.minecraft.init.Items;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.init.Items;
 import net.minecraft.item.ItemBlock;
```
### Errors:
- Unused import import net.minecraft.item.ItemBlock;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.init.Items;
 import net.minecraft.item.ItemBlock;
 import net.minecraft.item.ItemStack;
```
### Errors:
- Unused import import net.minecraft.item.ItemStack;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -12,6 +12,7 @@
+import net.minecraft.init.Items;
 import net.minecraft.item.ItemBlock;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
```
### Errors:
- Unused import import net.minecraft.nbt.NBTTagCompound;

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -163,7 +164,7 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
```
### Errors:
- Unused import import net.minecraft.nbt.NBTTagList;

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33885 |  5  |  2  |  +3  |
| Errors |  591  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10867&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
More formatting issues

More formatting issues, to compile with MC coding styles.

bob0bob commented:
updated to meet more of the coding style that MC wants.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33882 |  0  |  3  |  -3  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
Guards Request Armor #2577

This change will allow the guard to start requesting armor.  The type of armor will change based on the guards level.  From Leather - Diamond.  level 1 - 8.  At level 8, they stick with Diamond  armor.

Both the Ranger and Knight request the same armor type based on their level.

If you put Diamond armor in the inventory but they are requesting leather, they will only put on leather armor.

bob0bob committed:
Code formatting changes

code formatting changes to meet MC standards

bob0bob committed:
code changes for formatting

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.init.Items;
```
### Errors:
- Unused import import net.minecraft.init.Items;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.init.Items;
 import net.minecraft.init.SoundEvents;
```
### Errors:
- Unused import import net.minecraft.init.SoundEvents;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.init.Items;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.inventory.EntityEquipmentSlot;
```
### Errors:
- Unused import import net.minecraft.inventory.EntityEquipmentSlot;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -38,20 +39,22 @@
 import static com.minecolonies.coremod.entity.ai.citizen.guard.GuardConstants.*;
 import static com.minecolonies.coremod.entity.ai.util.AIState.*;
 
-@SuppressWarnings("squid:MaximumInheritanceDepth")
```
### Errors:
- Overly complex class AbstractEntityAIGuard (cyclomatic complexity = 103) 
- Class 'AbstractEntityAIGuard' has too many dependencies (26 > 10)
- Class 'AbstractEntityAIGuard' does not override 'toString()' method
- Class 'AbstractEntityAIGuard' has too many transitive dependencies (669 > 35)

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.init.Items;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
```
### Errors:
- Unused import import net.minecraft.item.Item;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.init.SoundEvents;
 import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemArmor;
```
### Errors:
- Unused import import net.minecraft.item.ItemArmor;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
 import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemArmor;
 import net.minecraft.item.ItemStack;
```
### Errors:
- Unused import import net.minecraft.item.ItemStack;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -25,6 +25,7 @@
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemArmor;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.ItemSword;
```
### Errors:
- Unused import import net.minecraft.item.ItemSword;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -38,20 +39,22 @@
```
### Errors:
- Unused import import net.minecraft.util.DamageSource;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -38,20 +39,22 @@
 
     /**
      * Tools and Items needed by the worker.
      */
```
### Errors:
- Can be package-private

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -38,20 +39,22 @@
     /**
      * Tools and Items needed by the worker.
      */
     public final List<ToolType>  toolsNeeded = new ArrayList<>();
```
### Errors:
- Can be package-private

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIKnight.java:
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.coremod.colony.jobs.JobKnight;
 import com.minecolonies.coremod.entity.ai.util.AIState;
 import net.minecraft.init.Items;
```
### Errors:
- Unused import import net.minecraft.init.Items;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIKnight.java:
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.coremod.colony.jobs.JobKnight;
 import com.minecolonies.coremod.entity.ai.util.AIState;
 import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
```
### Errors:
- Unused import import net.minecraft.inventory.EntityEquipmentSlot;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIKnight.java:
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.coremod.entity.ai.util.AIState;
 import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
 import net.minecraft.item.ItemStack;
```
### Errors:
- Unused import import net.minecraft.item.ItemStack;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRanger.java:
```diff
@@ -7,7 +7,6 @@
 
 import static com.minecolonies.coremod.entity.ai.util.AIState.*;
 
-@SuppressWarnings("squid:MaximumInheritanceDepth")
```
### Errors:
- Class 'EntityAIRanger' has too many transitive dependencies (669 > 35)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33917 |  5  |  9  |  -4  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10934&tab=Inspection&buildTypeId=bt88)

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33917 |  0  |  0  |  0   |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
Barracks - Show list all citizens assigned to that building #2597

This changes the default listing of hired workers for the building.  If there are more than one, then it creates a scroll bar with a list of all the workers assigned to this building.  This is really about the barracks where you can have up to 5 guards per barrack.

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/client/gui/AbstractWindowWorkerBuilding.java:
```diff
@@ -24,6 +26,18 @@
+    /**
+     * ScrollList with the workers.
+     */
+    private ScrollingList recipeList;
```
### Errors:
- Type of field 'recipeList' is concrete class ScrollingList 
- Instance field recipeList may not be initialized during object construction 
- Field can be converted to a local variable

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33927 | 13  |  3  | +10  |
| Errors |  591  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10960&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
code changes to meet formatting request.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33924 |  1  |  4  |  -3  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10980&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
Sleeping citizens - Cosmetic update, remove tool/weapon from hand while sleeping #2598

cosmetic change.  When a citizen goes to sleep.  Items in hand are removed.  They don't sleep hold a hoe/rishing rod, etc.....  Its just in the inventory.

bob0bob commented:
Fixed routine to remove missing sapling from loading.   Also,  fixed loading sapling loading for the building.  it was not loading them.
System is just merging all my pull requests into one pull request.  I can't have multiple from a different branch.  Or I can't figure out how to do it,  when I click on create new pull it dumps all changes into the pending one.

sorry for that.

bob0bob commented:
Sleeping citizens - Cosmetic update, remove tool/weapon from hand while sleeping #2598
Made changes so when citizens goes to sleep they remove item from hand. while sleeping.  Cosmetic only.

Raycoms commented on src/api/java/com/minecolonies/api/compatibility/CompatabilityManager.java:
```diff
@@ -163,7 +164,7 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
         for(final ItemStorage storage: storages)
         {
-            if(!saplings.contains(storage))
+            if(!saplings.contains(storage) && (storage.getItem() !=  Items.AIR))	//
```
can you remove this trailing comment here?

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java:
```diff
@@ -26,9 +30,17 @@
 import org.jetbrains.annotations.NotNull;
 
-import java.util.*;
+//import java.util.*;
```
I think we can remove this here as well

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java:
```diff
@@ -418,11 +470,64 @@ public void putInFurnace(final BlockPos currentFurnace, final BakingProduct baki
      */
     public static class View extends AbstractBuildingWorker.View
     {
+        private boolean[] recipesAllowed;
```
javadoc as well please

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingLumberjack.java:
```diff
@@ -222,6 +222,23 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
                 final boolean cut = saplingCompound.getBoolean(TAG_CUT);
                 treesToFell.put(new ItemStorage(stack), cut);
             }
+        } else {
```
wrong bracket formatting

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingLumberjack.java:
```diff
@@ -222,6 +222,23 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
+        	//We have a list of sapplings already, but we nee to load in the NBT file.
+        	//check to see if it exist and if so then update it.  Otherwise skip it,
+        	//as it doesn't exist in minecraft anymore.
+        	final NBTTagList saplingTagList = compound.getTagList(TAG_SAPLINGS, Constants.NBT.TAG_COMPOUND);
```
wrong formatting as well here

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/workorders/AbstractWorkOrder.java:
```diff
@@ -112,6 +113,7 @@ public static AbstractWorkOrder createFromNBT(@NotNull final NBTTagCompound comp
         }
         try
         {
+        	order.setPriority(compound.getInteger(TAG_PRIORITY));
```
formatting as well

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/BakerRecipes.java:
```diff
@@ -56,7 +59,15 @@
+
+    	
+    	
+    	final List<ItemStack> inputCake = new ArrayList<>();
```
formatting as well here

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java:
```diff
@@ -240,34 +238,37 @@ private AIState kneadTheDough()
-            getOwnBuilding().addToTasks(ProductState.PREPARED, currentBakingProduct);
-            currentBakingProduct = null;
-            return PREPARING;
+        	final IRecipeStorage storage = BakerRecipes.getRecipes().get(currentBakingProduct.getRecipeId());
```
the formatting looks off here as well

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java:
```diff
@@ -278,35 +279,54 @@ private AIState kneadTheDough()
+            building.addToTasks(bakingProduct.getState(), bakingProduct);
+            currentBakingProduct = bakingProduct;
+            return getState();
+        } else {
```
formatting

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/EntityAIWorkBaker.java:
```diff
@@ -317,42 +337,56 @@ private AIState createNewProduct()
+            currentBakingProduct.nextState();
+            getOwnBuilding().removeFromTasks(ProductState.UNCRAFTED, currentBakingProduct);
+            getOwnBuilding().addToTasks(ProductState.RAW, currentBakingProduct);
+        } else {
```
formatting

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -88,6 +91,26 @@ public AbstractEntityAIGuard(@NotNull final J job)
         worker.getCitizenExperienceHandler().setSkillModifier(2 * worker.getCitizenData().getStrength() + worker.getCitizenData().getIntelligence());
         worker.setCanPickUpLoot(true);
+        
+        itemsNeeded.add(new GuardItemsNeeded(EntityEquipmentSlot.FEET, Items.LEATHER_BOOTS, 1,2,3));
```
I think we can add this dynamically in the method getItemsNeeded depending on the level of the building?

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -149,9 +172,18 @@ else if (getOwnBuilding() != null)
+        for (final GuardItemsNeeded item : itemsNeeded)
         {
-            checkIfRequestForItemExistOrCreateAsynch(item);
+        	//Could have multiple armor request,  make sure the require armor falls into the guard
```
formatting seems off as well

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -603,4 +639,56 @@ private double getRandomPitch()
+    
+    
+    
+    public class GuardItemsNeeded
```
jdoc this class please and extract it into an own file

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33924 |  1  |  1  |  0   |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=10993&tab=Inspection&buildTypeId=bt88)

bob0bob committed:
More formatting changes to meets MC formatting style

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33919 |  1  |  6  |  -5  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=11012&tab=Inspection&buildTypeId=bt88)

bob0bob commented:
I'm not sure why it is showing all these changes still pending. I've submitted them.

Raycoms commented:
Hey, does the commit show in the commit history of the pull request?
https://github.com/ldtteam/minecolonies/pull/2593/commits

if not, did it push it?
Sometimes github borks and rejects a push.

LDTTeam-Buildsystem reviewed (approved):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33919 |  1  |  6  |  -5  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=11012&tab=Inspection&buildTypeId=bt88)

bob0bob commented:
not needed anymore.  To many changes got pulled into one PR.
